dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/datasource/AbstractDatasourceProcessor.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.api.utils.datasource;
+
+import org.apache.dolphinscheduler.api.dto.datasource.BaseDataSourceParamDTO;
+import org.apache.dolphinscheduler.api.dto.datasource.BaseHdfsDatasourceParamDTO;
+import org.apache.dolphinscheduler.api.enums.Status;
+import org.apache.dolphinscheduler.api.exceptions.ServiceException;
+import org.apache.dolphinscheduler.common.Constants;
+import org.apache.dolphinscheduler.common.enums.DbType;
+import org.apache.dolphinscheduler.common.utils.CommonUtils;
+import org.apache.dolphinscheduler.dao.datasource.MySQLDataSource;
+import org.apache.dolphinscheduler.spi.utils.JSONUtils;
+import org.apache.dolphinscheduler.spi.utils.StringUtils;
+
+import org.apache.commons.collections4.MapUtils;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public abstract class AbstractDatasourceProcessor implements DatasourceProcessor {
```

Comment:
The relationship between AbstractDatasourceProcessor and BaseDataSourceParamDTO and BaseDataSource

AbstractDatasourceProcessor and BaseDataSourceParamDTO are used to produce database parameters,

BaseDataSource is used to parse the parameters, it is recommended to consider BaseDataSource to modify together.

Reply:
Very good suggestion